### PR TITLE
add sincospi

### LINKED ISSUE #35799: add sincospi
Looking at #35792, I think it would make sense to also have a function `sincospi`, which could have a specialized implementation over just `(sinpi(x), cospi(x))`. It does look like there is some better performance to be gained.

### PR DESCRIPTION
**Pull request by simeonschaub**
close #35799
ref #35792

### CODE REVIEW

File: test/math.jl [hunk 1]
```diff
@@ -462,20 +470,25 @@ end
 
 @testset "Irrational args to sinpi/cospi/sinc/cosc" begin
     for x in (pi, ℯ, Base.MathConstants.golden)
-        @test sinpi(x) ≈ Float64(sinpi(big(x)))
-        @test cospi(x) ≈ Float64(cospi(big(x)))
+        for (sinpi, cospi) in ((sinpi, cospi), (x->sincospi(x)[1], x->sincospi(x)[2]))
+            @test sinpi(x) ≈ Float64(sinpi(big(x)))
+            @test cospi(x) ≈ Float64(cospi(big(x)))
+            @test sinpi(complex(x, x)) ≈ Complex{Float64}(sinpi(complex(big(x), big(x))))
+            @test cospi(complex(x, x)) ≈ Complex{Float64}(cospi(complex(big(x), big(x))))
```
Comment (simeonschaub) [inline test/math.jl:477]: These should really be tested a bit more

**Comment by StefanKarpinski** [comment]
This is great! I'm not qualified to review though. @simonbyrne, would you be willing?

**Comment by simonbyrne** [comment]
Can we factor out the range reduction so that `sinpi`/`cospi`/`sincospi` all use the same underlying function?

**Comment by simeonschaub** [comment]
I thought about this as well, but `sinpi` reduces to [-1,1], whereas `cospi` does [0,1], so they work a bit differently. One could perhaps use the same reduction for `sinpi` and `sincospi`, but I think factoring that out might add more complexity than is actually saved. Or do you think the deduplication is worth that complexity?

**Comment by simeonschaub** [comment]
@symonbyrne What kind of refactoring did you have in mind?

**Comment by simeonschaub** [comment]
Bump. What should be done here?

**Review by simonbyrne — APPROVED** [review]

**Comment by simeonschaub** [comment]
Does this mean this is good to go as-is?

**Comment by simonbyrne** [comment]
Yes, just wanted to figure out why the tests were failing.

**Pull request merged**